src/services/arweave/WarpDataProvider.ts

```diff
@@ -149,4 +158,87 @@ export class WarpDataProvider implements SmartweaveDataProvider {
       return error;
     }
   }
+
+  async handleTransactionExecution({
+    contractType,
+    interactionType,
+    transactionData,
```

Comment:
`transactionData` should have a type, and it can be unioined with `{ assetId: string` or assetId can be a separate flag on this field. regardless - should not be `any`

Reply:
changed